### ISSUE TITLE
Decide how Backend.copy_file() should handle sub-paths as destinations

### ISSUE THREAD

**Issue by hagenw** [audeering/audbackend#184]
As discussed in https://github.com/audeering/audbackend/pull/181#issuecomment-1933707596 and https://github.com/audeering/audbackend/pull/181#issuecomment-1933769671 `Backend.copy_file()` might behave in unexpected ways.

First, we should add tests for it to each backend.
And then we should decide if we would like to stay with possible different behavior that is then documented in the corresponding docstrings of the backends, or if we should provide a standard behavior, e.g. raising an error if a sub-path is given in all backends.

**Comment by frankenjoe** [comment]
As the example in https://github.com/audeering/audbackend/pull/181#issuecomment-1933769671 shows this is actually a generic issue and not specifically related to `Backend.copy_file()`.

**Comment by frankenjoe** [comment]
> provide a standard behavior, e.g. raising an error if a sub-path is given in all backends.

Possible, but probably costly since it involves a call to call `ls()`.

So maybe we leave it by adding to the documentation that a user must not put / copy / move a file to a sub-path of an existing file?

**Comment by hagenw** [comment]
Yes, it seems indeed a good idea to avoid adding some additional operations that first checks what exists and what not.
So, indeed updating the documentation seems the way to go here.
I would maybe not write that a user must not use an existing sub-path as destination, but mention that in this case we do not guarantee a fixed behavior as it depends on the underlying backend and platform. And hence recommend to not use a sub-path as destination.
